go/mobile/identity.go

```diff
@@ -0,0 +1,208 @@
+package iden3mobile
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	ethkeystore "github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/iden3/go-iden3-core/components/idenpuboffchain/readerhttp"
+	"github.com/iden3/go-iden3-core/components/idenpubonchain"
+	"github.com/iden3/go-iden3-core/db"
+	"github.com/iden3/go-iden3-core/eth"
+	"github.com/iden3/go-iden3-core/identity/holder"
+	babykeystore "github.com/iden3/go-iden3-core/keystore"
+	"github.com/iden3/go-iden3-crypto/babyjub"
+	log "github.com/sirupsen/logrus"
+)
+
+type Identity struct {
+	id          *holder.Holder
+	Tickets     *TicketsMap
+	eventSender Event
+	storage     db.Storage
+}
+
+// TODO:
+// store on changes vs store on stop
+
+// NewIdentity creates a new identity
+// this funciton is mapped as a constructor in Java.
+// NOTE: The storePath must be unique per Identity
+func NewIdentity(storePath, pass, web3Url string, checkTicketsPeriodMilis int, extraGenesisClaims *BytesArray, e Event) (*Identity, error) {
+	id := &Identity{}
+	idenPubOnChain, keyStore, storage, err := loadComponents(storePath, web3Url)
+	if err != nil {
+		return id, err
+	}
+	resourcesAreClosed := false
+	defer func() {
+		if !resourcesAreClosed {
+			keyStore.Close()
+			storage.Close()
+		}
+	}()
+	// Create babyjub keys
+	kOpComp, err := keyStore.NewKey([]byte(pass))
+	if err != nil {
+		return id, err
+	}
+	if err = keyStore.UnlockKey(kOpComp, []byte(pass)); err != nil {
+		return id, err
+	}
+	// Store kOpComp
+	tx, err := storage.NewTx()
+	if err != nil {
+		return id, err
+	}
+	if err := db.StoreJSON(tx, []byte("kOpComp"), kOpComp); err != nil {
```

Comment:
It's better to use constants for storage keys.  See here for example https://github.com/iden3/go-iden3-core/blob/master/identity/issuer/issuer.go#L32

Reply:
Fixing it